### DISCUSSION TITLE
Pull the latest code from the master branch and then build failed.

### DISCUSSION THREAD

**Discussion by whutpencil** [trinodb/trino#13511] [Q&A]
The key is that you can't see the error reporting from the construction log. Only the following contents are displayed: 

`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce (default) on project trino-pinot: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce (default) on project trino-pinot: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
`

I don't know how to solve it at all. How can I find the reason and solve it.

**Comment by hashhar** [comment]
If you search for "Rule" or "WARN" in the output you'll see the error message coming from the enforcer plugin.

I'm unable to reproduce this on the current master branch though.

**Comment by whutpencil** [comment]
> If you search for "Rule" or "WARN" in the output you'll see the error message coming from the enforcer plugin.
> 
> I'm unable to reproduce this on the current master branch though.

I did as you said and found the following information: `Failure to find org.apache.logging.log4j:log4j-core-java9:pom:2.11.2 in xxxx....`. Why is there a log4j dependency of java9? I can't find this package in my Maven image source. @hashhar

**Comment by ebyhr** [comment]
@whutpencil Could you share the result of `./mvnw --version`?

**Comment by ebyhr** [comment]
@whutpencil Please share the entire result.

**Comment by whutpencil** [comment]
> ./mvnw --version

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Users/xxxxxx/.m2/wrapper/dists/apache-maven-3.8.4-bin/5sjqtqkfjmakil1qhdtgrnjjdc/apache-maven-3.8.4
Java version: 11.0.13, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk-11.0.13.jdk/Contents/Home
Default locale: zh_CN_#Hans, platform encoding: UTF-8
OS name: "mac os x", version: "11.5", arch: "x86_64", family: "mac"

**Comment by ebyhr** [comment]
Additionally, please attache the entire log of `./mvnw clean install -DskipTests`.

**Comment by whutpencil** [comment]
> Additionally, please attache the entire log of `./mvnw clean install -DskipTests`.

OK, I'll do it again and export the log. Wait a minute.

**Comment by whutpencil** [comment]
[master_build_log.txt](https://github.com/trinodb/trino/files/8258646/master_build_log.txt)
cc @ebyhr

**Comment by ebyhr** [comment]
@whutpencil Are you using internal nexus or proxy?

**Comment by whutpencil** [comment]
> @whutpencil Are you using internal nexus or proxy?

Yes, I also added aliyun as the source.

**Comment by ebyhr** [comment]
How is your aliyun `mirror` node in your setting xml?

**Comment by whutpencil** [comment]
<mirror>
            <id>alimaven</id>
            <mirrorOf>central</mirrorOf>
            <name>aliyun maven</name>
            <url>https://maven.aliyun.com/repository/central/</url>
        </mirror>

**Comment by whutpencil** [comment]
> How is your aliyun `mirror` node in your setting xml?

Do you have a suitable source?

**Comment by wanhuhou** [comment]
> 

I have the same problem, did you solve it?
